vendor-overwrites/csslint/csslint.js

```diff
@@ -219,17 +227,51 @@ var CSSLint = (() => {
     },
   });
 
+  // Example 1:
+
+  /* csslint ignore:start */
+      // the chunk of code where errors won't be reported
+      // the chunk's start is hardwired to the next line after the opening comment
+      // the chunk's end is hardwired to the line of the closing comment
+  /* csslint ignore:end */
+
+  // Example 2:
+
+  /* csslint allow:rulename1,rulename2,... */
+      // allows to break the specified rules on the next single line of code
+
+  // Example 3:
+
+  /* csslint rulename1 */
+  /* csslint rulename2:N */
+  /* csslint rulename3:N, rulename4:N */
+
+      // entire code is affected;
+      // comments futher down the code extend/override previous comments of this kind
+      // values for N:
+      // "2" or "true" means "error"
+      // "1" or nothing means "warning" - note in this case ":" can also be omitted
+      // "0" or "false" means "ignore"
+      // (the quotes are added here for convenience, don't put them in the actual comments)
+
   function applyEmbeddedOverrides(text, ruleset, allow, ignore) {
     let ignoreStart = null;
     let ignoreEnd = null;
     let lineno = 0;
+    let eol = -1;
+    let m = RX_EMBEDDED.exec(text);
+    if (!m) return;
+
+    const nextLine = () => {
+      eol = (text.indexOf('\n', eol + 1) + 1 || text.length + 1) - 1;
+      lineno++;
+    };
 
-    for (let eol = 0, m; (m = RX_EMBEDDED.exec(text)); lineno++) {
-      eol = (text.indexOf('\n', eol) + 1 || text.length + 1) - 1;
-      if (eol < m.index) continue;
+    while (eol <= m.index) nextLine();
```

Comment:
I don't see the problem.
Is there a bug here or a performance impact?

Reply:
Yes, the line number isn't calculated correctly. For the following text:
```css
/* csslint ignore:start */
body {}
body {}
body {}
/* csslint ignore:end */
```
This ends up with an ignore range `[1, 2]`, because `lineno` always +1 when getting a new `RX_EMBEDDED` match.